server/controllers/list.js

```diff
@@ -569,29 +593,76 @@ const changeToOwner = (req, resp) => {
     });
 };
 
-const changeToMember = (req, resp) => {
+const removeOwnerRole = (req, resp) => {
   const { id: listId } = req.params;
   const { userId } = req.body;
+
   const {
     user: { _id: ownerId }
   } = req;
 
-  List.findOne({ _id: listId, ownerIds: { $all: [ownerId, userId] } })
+  List.findOne({ _id: listId, ownerIds: ownerId })
     .populate('cohortId', 'memberIds ownerIds')
     .exec()
+    .then(doc => {
+      if (!doc) {
+        throw new BadRequestException("Can't remove owner role.");
+      }
+
+      const userIsOwner = doc.ownerIds.some(id => id.equals(userId));
+
+      if (userIsOwner) {
+        doc.ownerIds.splice(doc.ownerIds.indexOf(userId), 1);
+      }
+
+      return doc.save();
+    })
+    .then(() =>
+      resp.status(200).send({
+        message: 'User has no owner role.'
+      })
+    )
+    .catch(err => {
+      if (err instanceof BadRequestException) {
+        const { status, message } = err;
+        return resp.status(status).send({ message });
+      }
+
+      resp.status(400).send({ message: 'List data not found' });
+    });
+};
+
+const addMemberRole = (req, resp) => {
+  const { id: listId } = req.params;
+  const { userId } = req.body;
+  const {
+    user: { _id: ownerId }
+  } = req;
+
+  List.findOne({
+    _id: listId,
+    ownerIds: { $in: [ownerId] }
+  })
+    .populate('cohortId', 'memberIds ownerIds viewersIds')
+    .exec()
     .then(doc => {
       if (!doc) {
         throw new BadRequestException("Can't set user as a list's member.");
       }
 
-      const { cohortId: cohort, isPrivate, ownerIds } = doc;
-      const isNotCohortMember = !checkIfCohortMember(cohort, userId);
+      const { ownerIds, memberIds } = doc;
+      const userIsNotAMember = !memberIds.some(id => id.equals(userId));
 
-      if (isPrivate || isNotCohortMember) {
-        doc.memberIds.push(userId);
+      if (userIsNotAMember) {
+        memberIds.push(userId);
+      }
+
+      const userIsOwner = ownerIds.some(id => id.equals(userId));
```

Comment:
You wrote separate helper for that checking. Couldn't be used here? There more place where it could be used...